### DISCUSSION TITLE
More options for server management on User side

### DISCUSSION THREAD

**Discussion by njmdev03** [pelican-dev/panel#27] [Ideas]
My biggest frustration with Ptero (and by extension Pelican) is that users cannot modify alot of aspects of their server. For example, a user cannot change their servers egg to switch their server from Vanilla MC to Curseforge MC. This is a relatively common flow, and currently it requires switching between the user and admin panels.

Another nice feature for the user front-end would be to allow an admin to give users an amount of resources that they can alocate to servers themselves. This is less common in commercial settings, but I have seen hosts do it, and it would be very nice for home users.

**Comment by martinbndr** [comment]
Yea also thought on this, being able to change the "subversion" of an egg inside the server should be possible
Also self creating servers in the user front end (with optional limited resources) seems nice.

But I think it should be all optionally (configureable in the admin panel) since there are usecases servers should only be created / egg modified in a protected way.

**Comment by lancepioch** [comment]
Thanks, this is in the works, might take some time though!
